### PR TITLE
[+] Docker - Add Docker build files

### PR DESCRIPTION
**Pull request by root-io**

### CODE REVIEW

**Comment by root-io** [comment]
[Lumber - Add Docker build files](https://trello.com/c/DG0RJnrg/3291-lumber-add-docker-build-files)

File: templates/app/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: '3'
+services:
+  app:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    container_name: <%= appName %>
+    ports:
+      - "<%= port %>:<%= port %>"
```
Comment (ArnaudValensi) [inline templates/app/docker-compose.yaml]: To be more standard and avoid the user to know how forest handle environment variable, I would not copy the .env file and configure the variable directly in the docker-compose.yaml.

File: services/dumper.js [hunk 2]
```diff
@@ -174,6 +174,44 @@ function Dumper(project, config) {
...
+      databaseUrl: getDatabaseUrl(),
+      forestUrl: process.env.FOREST_URL,
+      devRenderingId: project.defaultEnvironment.renderings[0].id,
+      ssl: config.ssl,
+      encrypt: config.ssl && config.dbDialect === 'mssql',
+      dbSchema: config.dbSchema,
+      port: config.appPort,
+    };
+
+    fs.writeFileSync(`${pathDest}/docker-compose.yaml`, template(settings));
```
Comment (SeyZ) [inline services/dumper.js]: It's `docker-compose.yml`, not `yaml`

File: templates/app/docker-compose.yaml (hunk 1 above)
Comment (SeyZ) [reply templates/app/docker-compose.yaml]: Me too!

File: templates/app/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+COPY package*.json ./
+
+RUN npm install
+
+COPY . .
+
+EXPOSE <%= port %>
+
+CMD ["npm", "start"]
```
Comment (SeyZ) [inline templates/app/Dockerfile]: Can we remove the empty lines between each line?

File: services/dumper.js (hunk 2 above)
Comment (root-io) [reply services/dumper.js]: From the doc it seems we can use both:

https://docs.docker.com/compose/compose-file/#service-configuration-reference

`Tip: You can use either a .yml or .yaml extension for this file. They both work.`

File: templates/app/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      context: .
+      dockerfile: Dockerfile
+    container_name: <%= appName %>
+    environment:
+      - DATABASE_URL=<%= databaseUrl %>
+      - FOREST_AUTH_SECRET=<%= forestAuthSecret %>
+      - FOREST_ENV_SECRET=<%= forestEnvSecret %>
+      <% if (!ssl) {%># <% } %>- SSL_DATABASE=true
+      <% if (!encrypt) {%># <% } %>- ENCRYPT_DATABASE=true
+      <% if (forestUrl) {%>- FOREST_URL=<%= forestUrl %><% } %>
```
Comment (ArnaudValensi) [inline templates/app/docker-compose.yml]: We can remove this.

File: templates/app/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      <% if (devRenderingId) {%>- DEV_RENDERING_ID=<%= devRenderingId %><% } %>
```
Comment (ArnaudValensi) [inline templates/app/docker-compose.yml]: We can remove this too.

File: templates/app/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      <% if (dbSchema) {%>- DATABASE_SCHEMA=<%= dbSchema %><% } %>
```
Comment (ArnaudValensi) [inline templates/app/docker-compose.yml]: Keep this but remove the condition.

File: templates/app/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+      <% if (port) {%>- PORT=<%= port %><% } %>
```
Comment (ArnaudValensi) [inline templates/app/docker-compose.yml]: We can remove this too.

**Pull request merged**